internal/pkg/cli/task_delete.go

```diff
@@ -0,0 +1,380 @@
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"github.com/aws/copilot-cli/internal/pkg/term/selector"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	taskDeleteNamePrompt              = "Which task would you like to delete?"
+	taskDeleteAppPrompt               = "Which application would you like to delete a task from?"
+	taskDeleteEnvPrompt               = "Which environment would you like to delete a task from?"
+	fmtTaskDeleteDefaultConfirmPrompt = "Are you sure you want to delete %s from the default cluster?"
+	fmtTaskDeleteFromEnvConfirmPrompt = "Are you sure you want to delete %s from application %s and environment %s?"
+	taskDeleteConfirmHelp             = "This will delete the task's stack and stop all current executions."
+)
+
+var errTaskDeleteCancelled = errors.New("task delete cancelled - no changes made")
+
+type deleteTaskVars struct {
+	name             string
+	app              string
+	env              string
+	skipConfirmation bool
+	defaultCluster   bool
+}
+
+type deleteTaskOpts struct {
+	deleteTaskVars
+
+	// Dependencies to interact with other modules
+	store  store
+	prompt prompter
+	// spinner progress
+	sess sessionProvider
+	sel  wsSelector
+
+	// Generators for env-specific clients
+	newTaskSel func(session *awssession.Session) cfTaskSelector
+	// newTaskStopper  func(session *awssession.Session) tasksStopper
+	// newTaskLister   func(session *awssession.Session) tasksLister
+	// newImageRemover func(session *awssession.Session) imageRemover
+	// newTaskDeleter  func(session *awssession.Session) taskDeployer
+}
+
+func newDeleteTaskOpts(vars deleteTaskVars) (*deleteTaskOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("new config store: %w", err)
+	}
+
+	provider := sessions.NewProvider()
+
+	prompter := prompt.New()
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, fmt.Errorf("new workspace: %w", err)
+	}
+
+	return &deleteTaskOpts{
+		deleteTaskVars: vars,
+
+		store: store,
+		// spinner: termprogress.NewSpinner(log.DiagnosticWriter),
+		prompt: prompter,
+		sess:   provider,
+		sel:    selector.NewWorkspaceSelect(prompter, store, ws),
+		newTaskSel: func(session *awssession.Session) cfTaskSelector {
+			cfn := cloudformation.New(session)
+			return selector.NewCFTaskSelect(prompter, store, cfn)
+		},
+		// newTaskLister: func(session *awssession.Session) tasksLister {
+		// 	return ecs.New(session)
+		// },
+		// newTaskStopper: func(session *awssession.Session) tasksStopper {
+		// 	return ecs.New(session)
+		// },
+		// newTaskDeleter: func(session *awssession.Session) taskDeployer {
+		// 	return cloudformation.New(session)
+		// },
+		// newImageRemover: func(session *awssession.Session) imageRemover {
+		// 	return ecr.New(session)
+		// },
+	}, nil
+}
+
+// Validate checks that flag inputs are valid.
+func (o *deleteTaskOpts) Validate() error {
+
+	if o.name != "" {
```

Comment:
I'm probably missing it-- do we validate that the flag-inputted task exists?

Reply:
No. It's difficult because in the case where a task is provided but no app, env, or default cluster flag, we don't know where to look for the task. 

In these cases we rely on error messages returned when we go looking for the task or service in the `Execute` logic.